src/internal/connector/onedrive/collections.go

```diff
@@ -230,8 +230,15 @@ func (c *Collections) UpdateCollections(
 			// already created and partially populated.
 			updatePath(newPaths, *item.GetId(), folderPath.String())
 
+			if c.source != OneDriveSource {
+				continue
+			}
+
+			fallthrough
```

Comment:
ah, I think I see. I'm guessing the reason permissions for a folder are currently in the parent and not the folder itself is so that we already have them when we need to make the folder during restore?

I guess the question for whether it would be "easy" to keep folder permissions in the folder itself is really whether there is an ordering issue between restoring item permissions and folder permissions. If I remember right, you did say there was a potential ordering issue and that folder permissions should be restored first right?

Reply:
I initially had it within the folder but moved out mainly because of the ordering issue. If we keep the `.dirmeta` file inside the folder we have to make sure we are fetching that file first, but this way we can be certain that by the time a folder contents is to be restored we have created the folder with the right permissions.